hal-sim/hal_impl/functions.py

```diff
@@ -9,6 +9,7 @@
 from hal_impl.sim_hooks import SimHooks
 
 import logging
+from hal_impl.mode_helpers import notify_new_ds_data
```

Comment:
This creates an import loop. This completely breaks on Python 3.4.

Reply:
Ahh that’s why that broke